rqt_gui_py/src/rqt_gui_py/rclpy_spinner.py

```diff
@@ -30,8 +30,6 @@ def run(self):
         executor.add_node(self._node)
         while rclpy.ok() and not self._abort:
             executor.spin_once(timeout_sec=1.0)
-        if not self._abort:
```

Comment:
I'm not sure if we should remove this warning since we haven't really fixed the underlying problem described [here](https://github.com/ros2/rclpy/issues/1081#issuecomment-1542164408)

But the switch to `try_shutdown()` resolves the "rcl_shutdown" aleady called.." error from being logged.

Reply:
The [general strategy](https://github.com/ros2/rclpy/issues/1081#issuecomment-1542164408) seems good to me.

Let me explain what I see while debugging the `rqt_graph` and why I remove the warning message.

<details><summary> gdb for rqt_graph when ctrl+c </summary>
<p>

```shell
(ros2 source rolling setup) domain=28 RMW= 
[15:53:42 OptiPlex-7080] /home/chenlh/Projects/ROS2/ros2-master/install  
chenlh install $ gdb /usr/bin/python3
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/python3...
Reading symbols from /usr/lib/debug/.build-id/14/8e086667839ef13939196984d6f717c331bd76.debug...
(gdb) handle SIGINT stop pass
SIGINT is used by the debugger.
Are you sure you want to change it? (y or n) y
Signal        Stop	Print	Pass to program	Description
SIGINT        Yes	Yes	Yes		Interrupt
(gdb) b rcl_shutdown
Function "rcl_shutdown" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (rcl_shutdown) pending.
(gdb) r /home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph
Starting program: /usr/bin/python3 /home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe9c3e640 (LWP 2124744)]
[New Thread 0x7fffe3fff640 (LWP 2124745)]
[New Thread 0x7fffe37fe640 (LWP 2124746)]
[New Thread 0x7fffe2ffd640 (LWP 2124747)]
[New Thread 0x7fffe27fc640 (LWP 2124748)]
[New Thread 0x7fffe1ffb640 (LWP 2124749)]
[New Thread 0x7fffe0ff8640 (LWP 2124756)]
[New Thread 0x7fffcbfff640 (LWP 2124757)]
[New Thread 0x7fffcb7fe640 (LWP 2124758)]
[New Thread 0x7fffcaffd640 (LWP 2124759)]
[New Thread 0x7fffca7fc640 (LWP 2124760)]
[New Thread 0x7fffc9ffb640 (LWP 2124761)]
[New Thread 0x7fffc97fa640 (LWP 2124762)]
[New Thread 0x7fffc8ff9640 (LWP 2124763)]
[New Thread 0x7fffabfff640 (LWP 2124764)]
[New Thread 0x7fffab7fe640 (LWP 2124765)]
[New Thread 0x7fffaaffd640 (LWP 2124766)]
[New Thread 0x7fffaa7fc640 (LWP 2124767)]
[New Thread 0x7fffa9ffb640 (LWP 2124768)]
[New Thread 0x7fffa97fa640 (LWP 2124769)]
[New Thread 0x7fffa8ff9640 (LWP 2124770)]
[New Thread 0x7fff8bfff640 (LWP 2124771)]
[New Thread 0x7fff88dff640 (LWP 2124895)]
[New Thread 0x7fff7ffff640 (LWP 2124896)]
[New Thread 0x7fff7f7fe640 (LWP 2124897)]
[New Thread 0x7fff7effd640 (LWP 2124898)]
[New Thread 0x7fff7e7fc640 (LWP 2124899)]
[New Thread 0x7fff7dffb640 (LWP 2124900)]
[New Thread 0x7fff7d7fa640 (LWP 2124901)]
[New Thread 0x7fff7cff9640 (LWP 2124902)]
[New Thread 0x7fff7c7f8640 (LWP 2124903)]
[New Thread 0x7fff7bff7640 (LWP 2124904)]
[New Thread 0x7fff7b7f6640 (LWP 2124905)]
[New Thread 0x7fff7aff5640 (LWP 2124911)]
[New Thread 0x7fff79fff640 (LWP 2125041)]
[Detaching after vfork from child process 2125069]
[New Thread 0x7fff797fe640 (LWP 2125082)]
[New Thread 0x7fff78ffd640 (LWP 2125083)]
[New Thread 0x7fff5b5ff640 (LWP 2125088)]
[New Thread 0x7fff5adfe640 (LWP 2125089)]
[New Thread 0x7fff5a5fd640 (LWP 2125090)]
[New Thread 0x7fff59dfc640 (LWP 2125091)]
[New Thread 0x7fff595fb640 (LWP 2125092)]
[New Thread 0x7fff58dfa640 (LWP 2125093)]
[New Thread 0x7fff3bfff640 (LWP 2125094)]
[New Thread 0x7fff3b7fe640 (LWP 2125095)]
[New Thread 0x7fff3affd640 (LWP 2125097)]
[New Thread 0x7fff3a7fc640 (LWP 2125099)]
[New Thread 0x7fff39ffb640 (LWP 2125100)]
^C
Thread 1 "python3" received signal SIGINT, Interrupt.
0x00007ffff7d18d7f in __GI___poll (fds=0x55555723c790, nfds=6, timeout=524) at ../sysdeps/unix/sysv/linux/poll.c:29
29	../sysdeps/unix/sysv/linux/poll.c: No such file or directory.
(gdb) c
Continuing.
[INFO] [1684223676.475721938] [rclcpp]: signal_handler(signum=2)
[Switching to Thread 0x7fff7f7fe640 (LWP 2124897)]

Thread 26 "python3" hit Breakpoint 1, rcl_shutdown (context=0x55555667d650) at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/init.c:305
305	{
(gdb) bt
#0  rcl_shutdown (context=0x55555667d650) at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/init.c:305
#1  0x00007ffff6579f7b in rclpy::shutdown_contexts() () at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclpy/rclpy/src/rclpy/context.cpp:49
#2  0x00007ffff6630a83 in operator()() const (__closure=0x5555564e3448)
    at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclpy/rclpy/src/rclpy/signal_handler.cpp:144
#3  0x00007ffff66335fc in std::__invoke_impl<void, (anonymous namespace)::setup_deferred_signal_handler()::<lambda()> >(std::__invoke_other, struct {...} &&) (__f=...) at /usr/include/c++/11/bits/invoke.h:61
#4  0x00007ffff66335bf in std::__invoke<(anonymous namespace)::setup_deferred_signal_handler()::<lambda()> >(struct {...} &&) (__fn=...)
    at /usr/include/c++/11/bits/invoke.h:96
#5  0x00007ffff663356c in std::thread::_Invoker<std::tuple<(anonymous namespace)::setup_deferred_signal_handler()::<lambda()> > >::_M_invoke<0>(std::_Index_tuple<0>) (this=0x5555564e3448) at /usr/include/c++/11/bits/std_thread.h:253
#6  0x00007ffff6633540 in std::thread::_Invoker<std::tuple<(anonymous namespace)::setup_deferred_signal_handler()::<lambda()> > >::operator()(void)
    (this=0x5555564e3448) at /usr/include/c++/11/bits/std_thread.h:260
#7  0x00007ffff6633524 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<(anonymous namespace)::setup_deferred_signal_handler()::<lambda()> > > >::_M_run(void) (this=0x5555564e3440) at /usr/include/c++/11/bits/std_thread.h:211
#8  0x00007ffff5adc2b3 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x00007ffff7c94b43 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#10 0x00007ffff7d26a00 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
(gdb) c
Continuing.
[Switching to Thread 0x7fff7ffff640 (LWP 2124896)]

Thread 25 "python3" hit Breakpoint 1, rcl_shutdown (context=0x555556646b00) at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/init.c:305
305	{
(gdb) bt
#0  rcl_shutdown (context=0x555556646b00) at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/init.c:305
#1  0x00007fff8b246f54 in rclcpp::Context::shutdown(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    (this=0x55555614def0, reason="signal handler") at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/src/rclcpp/context.cpp:321
#2  0x00007fff8b429187 in rclcpp::SignalHandler::deferred_signal_handler()
    (this=0x7fff8b69d320 <rclcpp::SignalHandler::get_global_signal_handler()::signal_handler>)
    at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclcpp/rclcpp/src/rclcpp/signal_handler.cpp:271
#3  0x00007fff8b42a513 in std::__invoke_impl<void, void (rclcpp::SignalHandler::*)(), rclcpp::SignalHandler*>(std::__invoke_memfun_deref, void (rclcpp::SignalHandler::*&&)(), rclcpp::SignalHandler*&&)
    (__f=@0x5555565c4e20: (void (rclcpp::SignalHandler::*)(class rclcpp::SignalHandler * const)) 0x7fff8b428d68 <rclcpp::SignalHandler::deferred_signal_handler()>, __t=@0x5555565c4e18: 0x7fff8b69d320 <rclcpp::SignalHandler::get_global_signal_handler()::signal_handler>)
    at /usr/include/c++/11/bits/invoke.h:74
#4  0x00007fff8b42a495 in std::__invoke<void (rclcpp::SignalHandler::*)(), rclcpp::SignalHandler*>(void (rclcpp::SignalHandler::*&&)(), rclcpp::SignalHandler*&&)
    (__fn=@0x5555565c4e20: (void (rclcpp::SignalHandler::*)(class rclcpp::SignalHandler * const)) 0x7fff8b428d68 <rclcpp::SignalHandler::deferred_signal_handler()>) at /usr/include/c++/11/bits/invoke.h:96
#5  0x00007fff8b42a3f5 in std::thread::_Invoker<std::tuple<void (rclcpp::SignalHandler::*)(), rclcpp::SignalHandler*> >::_M_invoke<0ul, 1ul>(std::_Index_tuple<0ul, 1ul>) (this=0x5555565c4e18) at /usr/include/c++/11/bits/std_thread.h:253
#6  0x00007fff8b42a3aa in std::thread::_Invoker<std::tuple<void (rclcpp::SignalHandler::*)(), rclcpp::SignalHandler*> >::operator()()
    (this=0x5555565c4e18) at /usr/include/c++/11/bits/std_thread.h:260
#7  0x00007fff8b42a38a in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (rclcpp::SignalHandler::*)(), rclcpp::SignalHandler*> > >::_M_run() (this=0x5555565c4e10) at /usr/include/c++/11/bits/std_thread.h:211
#8  0x00007ffff5adc2b3 in  () at /lib/x86_64-linux-gnu/libstdc++.so.6
#9  0x00007ffff7c94b43 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#10 0x00007ffff7d26a00 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
(gdb) c
Continuing.
[Thread 0x7fffe37fe640 (LWP 2124746) exited]
[Thread 0x7fff79fff640 (LWP 2125041) exited]
[Thread 0x7fff7aff5640 (LWP 2124911) exited]
[Thread 0x7fff7e7fc640 (LWP 2124899) exited]
[Thread 0x7fff7dffb640 (LWP 2124900) exited]
[Thread 0x7fff7d7fa640 (LWP 2124901) exited]
[Thread 0x7fff7cff9640 (LWP 2124902) exited]
[Thread 0x7fff7c7f8640 (LWP 2124903) exited]
[Thread 0x7fff7bff7640 (LWP 2124904) exited]
[Thread 0x7fff7b7f6640 (LWP 2124905) exited]
[Switching to Thread 0x7ffff7e4b1c0 (LWP 2124634)]

Thread 1 "python3" hit Breakpoint 1, rcl_shutdown (context=0x55555667d650) at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/init.c:305
305	{
(gdb) bt
#0  rcl_shutdown (context=0x55555667d650) at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rcl/rcl/src/rcl/init.c:305
#1  0x00007ffff657aac7 in rclpy::Context::shutdown() (this=0x555556670cd0)
    at /home/chenlh/Projects/ROS2/ros2-master/src/ros2/rclpy/rclpy/src/rclpy/context.cpp:161
#2  0x00007ffff657e2f3 in pybind11::cpp_function::cpp_function<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}::operator()(rclpy::Context*) const (__closure=0x555555db7e18, c=0x555556670cd0) at /usr/include/pybind11/pybind11.h:108
#3  0x00007ffff658160b in pybind11::detail::argument_loader<rclpy::Context*>::call_impl<void, pybind11::cpp_function::cpp_function<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}&, 0ul, pybind11::detail::void_type>(pybind11::cpp_function::cpp_function<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}&, std::integer_sequence<unsigned long, 0ul>, pybind11::detail::void_type&&) && (this=0x7fffffff7020, f=...) at /usr/include/pybind11/cast.h:1207
#4  0x00007ffff658072e in pybind11::detail::argument_loader<rclpy::Context*>::call<void, pybind11::detail::void_type, pybind11::cpp_function::cpp_function<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}&>(pybind11::cpp_function::cpp_function<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}&) && (this=0x7fffffff7020, f=...) at /usr/include/pybind11/cast.h:1184
#5  0x00007ffff657f787 in pybind11::cpp_function::initialize<pybind11::cpp_function::initialize<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}, void, rclpy::Context*, pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(pybind11::cpp_function::initialize<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}&&, void (*)(rclpy::Context*), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(pybind11::detail::function_call&)#3}::operator()(pybind11::detail::function_call&) const (__closure=0x0, call=...) at /usr/include/pybind11/pybind11.h:233
#6  0x00007ffff657f7ef in pybind11::cpp_function::initialize<pybind11::cpp_function::initialize<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}, void, rclpy::Context*, pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(pybind11::cpp_function::initialize<void, rclpy::Context, , pybind11::name, pybind11::is_method, pybind11::sibling, char [17]>(void (rclpy::Context::*)(), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(rclpy::Context*)#1}&&, void (*)(rclpy::Context*), pybind11::name const&--Type <RET> for more, q to quit, c to continue without paging--c
, pybind11::is_method const&, pybind11::sibling const&, char const (&) [17])::{lambda(pybind11::detail::function_call&)#3}::_FUN(pybind11::detail::function_call&) () at /usr/include/pybind11/pybind11.h:210
#7  0x00007ffff64da600 in pybind11::cpp_function::dispatcher(_object*, _object*, _object*) (self=<PyCapsule at remote 0x7ffff696ba20>, args_in=(<rclpy._rclpy_pybind11.Context at remote 0x7fffe0367fb0>,), kwargs_in=0x0) at /usr/include/pybind11/pybind11.h:835
#8  0x00005555556b099e in cfunction_call (func=<built-in method shutdown of PyCapsule object at remote 0x7ffff696ba20>, args=<optimized out>, kwargs=<optimized out>) at ../Objects/methodobject.c:543
#9  0x00005555556a74ab in _PyObject_MakeTpCall (tstate=0x555555b67000, callable=<built-in method shutdown of PyCapsule object at remote 0x7ffff696ba20>, args=<optimized out>, nargs=<optimized out>, keywords=0x0) at ../Objects/call.c:215
#10 0x00005555556bef0b in _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571feb80, callable=<built-in method shutdown of PyCapsule object at remote 0x7ffff696ba20>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:112
#11 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571feb80, callable=<built-in method shutdown of PyCapsule object at remote 0x7ffff696ba20>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:99
#12 method_vectorcall (method=<optimized out>, args=0x5555571feb88, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#13 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571feb88, callable=<method at remote 0x7fffea01bf00>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#14 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571feb88, callable=<method at remote 0x7fffea01bf00>) at ../Include/cpython/abstract.h:123
#15 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff7640, tstate=<optimized out>) at ../Python/ceval.c:5891
#16 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#17 0x00005555556b11ec in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571fea00, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/rclpy/context.py, line 112, in shutdown (self=<Context(_lock=<_thread.lock at remote 0x7ffff5896400>, _callbacks=[], _logging_initialized=True, _Context__context=<rclpy._rclpy_pybind11.Context at remote 0x7fffe0367fb0>) at remote 0x7fffe03c0c70>), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#18 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, con=0x7ffff6a8b0b0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#19 _PyFunction_Vectorcall (func=<function at remote 0x7ffff6a8b0a0>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/call.c:342
#20 0x00005555556998cb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f5d50, callable=<function at remote 0x7ffff6a8b0a0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#21 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f5d50, callable=<function at remote 0x7ffff6a8b0a0>) at ../Include/cpython/abstract.h:123
#22 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff7810, tstate=<optimized out>) at ../Python/ceval.c:5891
#23 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4198
#24 0x00005555556b11ec in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571f5be0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/rclpy/utilities.py, line 82, in shutdown (context=<Context(_lock=<_thread.lock at remote 0x7ffff5896400>, _callbacks=[], _logging_initialized=True, _Context__context=<rclpy._rclpy_pybind11.Context at remote 0x7fffe0367fb0>) at remote 0x7fffe03c0c70>), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#25 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, con=0x7ffff69797f0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#26 _PyFunction_Vectorcall (func=<function at remote 0x7ffff69797e0>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/call.c:342
#27 0x000055555569aadf in _PyObject_VectorcallTstate (kwnames=('context',), nargsf=<optimized out>, args=<optimized out>, callable=<function at remote 0x7ffff69797e0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#28 PyObject_Vectorcall (kwnames=('context',), nargsf=<optimized out>, args=<optimized out>, callable=<function at remote 0x7ffff69797e0>) at ../Include/cpython/abstract.h:123
#29 call_function (kwnames=('context',), oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff79e0, tstate=<optimized out>) at ../Python/ceval.c:5891
#30 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4231
#31 0x00005555556b11ec in _PyEval_EvalFrame (throwflag=0, f=Frame 0x555557206580, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/rclpy/__init__.py, line 126, in shutdown (context=None, uninstall_handlers=None), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#32 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, con=0x7ffff697aba0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#33 _PyFunction_Vectorcall (func=<function at remote 0x7ffff697ab90>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/call.c:342
#34 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571fe6b8, callable=<function at remote 0x7ffff697ab90>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#35 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571fe6b8, callable=<function at remote 0x7ffff697ab90>) at ../Include/cpython/abstract.h:123
#36 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff7bb0, tstate=<optimized out>) at ../Python/ceval.c:5891
#37 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#38 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571fe540, for file /home/chenlh/Projects/ROS2/ros2-master/build/rqt_gui_py/src/rqt_gui_py/ros_py_plugin_provider.py, line 89, in _destroy_node (self=<RosPyPluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='rqt_gui', _base_class_type='rqt_gui_py::Plugin', _plugin_descriptors={'rqt_action/ActionPlugin': <PluginDescriptor(_dict={'plugin_id': 'rqt_action/ActionPlugin', 'attributes': {'package_name': 'rqt_action', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_action', 'library_path': 'src', 'class_name': 'ActionPlugin', 'class_type': 'rqt_action.action_plugin.ActionPlugin', 'class_base_class_type': 'rqt_gui_py::Plugin', 'plugin_id': 'rqt_action/ActionPlugin', 'module_name': 'rqt_action.action_plugin', 'class_from_class_type': 'ActionPlugin', 'not_available': ''}, 'action': {'label': 'Action Type Browser', 'statustip': 'A Python GUI for browsing available ROS action types.', 'icon': 'mail-read', 'icontype': 'theme'}, 'groups': [{'label': 'Actions', 'statustip...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#39 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571e2cf0, locals=0x0, con=0x7fffe03944d0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#40 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571e2cf0, func=<function at remote 0x7fffe03944c0>) at ../Objects/call.c:342
#41 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571e2cf0, callable=<function at remote 0x7fffe03944c0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#42 method_vectorcall (method=<optimized out>, args=0x5555571e2cf8, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#43 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571e2cf8, callable=<method at remote 0x7fffc80ae880>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#44 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571e2cf8, callable=<method at remote 0x7fffc80ae880>) at ../Include/cpython/abstract.h:123
#45 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff7de0, tstate=<optimized out>) at ../Python/ceval.c:5891
#46 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#47 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571e2b70, for file /home/chenlh/Projects/ROS2/ros2-master/build/rqt_gui_py/src/rqt_gui_py/ros_py_plugin_provider.py, line 62, in shutdown (self=<RosPyPluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='rqt_gui', _base_class_type='rqt_gui_py::Plugin', _plugin_descriptors={'rqt_action/ActionPlugin': <PluginDescriptor(_dict={'plugin_id': 'rqt_action/ActionPlugin', 'attributes': {'package_name': 'rqt_action', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_action', 'library_path': 'src', 'class_name': 'ActionPlugin', 'class_type': 'rqt_action.action_plugin.ActionPlugin', 'class_base_class_type': 'rqt_gui_py::Plugin', 'plugin_id': 'rqt_action/ActionPlugin', 'module_name': 'rqt_action.action_plugin', 'class_from_class_type': 'ActionPlugin', 'not_available': ''}, 'action': {'label': 'Action Type Browser', 'statustip': 'A Python GUI for browsing available ROS action types.', 'icon': 'mail-read', 'icontype': 'theme'}, 'groups': [{'label': 'Actions', 'statustip': 'P...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#48 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571fb380, locals=0x0, con=0x7fffe0394290, tstate=0x555555b67000) at ../Python/ceval.c:5065
#49 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571fb380, func=<function at remote 0x7fffe0394280>) at ../Objects/call.c:342
#50 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571fb380, callable=<function at remote 0x7fffe0394280>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#51 method_vectorcall (method=<optimized out>, args=0x5555571fb388, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#52 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571fb388, callable=<method at remote 0x7fff8854ea40>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#53 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571fb388, callable=<method at remote 0x7fff8854ea40>) at ../Include/cpython/abstract.h:123
#54 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff8010, tstate=<optimized out>) at ../Python/ceval.c:5891
#55 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#56 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571fb200, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/composite_plugin_provider.py, line 87, in shutdown (self=<RecursivePluginProvider(_plugin_providers=[<RosPyPluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='rqt_gui', _base_class_type='rqt_gui_py::Plugin', _plugin_descriptors={'rqt_action/ActionPlugin': <PluginDescriptor(_dict={'plugin_id': 'rqt_action/ActionPlugin', 'attributes': {'package_name': 'rqt_action', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_action', 'library_path': 'src', 'class_name': 'ActionPlugin', 'class_type': 'rqt_action.action_plugin.ActionPlugin', 'class_base_class_type': 'rqt_gui_py::Plugin', 'plugin_id': 'rqt_action/ActionPlugin', 'module_name': 'rqt_action.action_plugin', 'class_from_class_type': 'ActionPlugin', 'not_available': ''}, 'action': {'label': 'Action Type Browser', 'statustip': 'A Python GUI for browsing available ROS action types.', 'icon': 'mail-read', 'icontype'...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#57 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571f90a0, locals=0x0, con=0x7ffff43ca210, tstate=0x555555b67000) at ../Python/ceval.c:5065
#58 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571f90a0, func=<function at remote 0x7ffff43ca200>) at ../Objects/call.c:342
#59 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571f90a0, callable=<function at remote 0x7ffff43ca200>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#60 method_vectorcall (method=<optimized out>, args=0x5555571f90a8, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#61 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f90a8, callable=<method at remote 0x7fffea019100>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#62 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f90a8, callable=<method at remote 0x7fffea019100>) at ../Include/cpython/abstract.h:123
#63 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff8240, tstate=<optimized out>) at ../Python/ceval.c:5891
#64 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#65 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571f8f20, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/composite_plugin_provider.py, line 87, in shutdown (self=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-python', 'icontype': 'theme'}, 'groups': [{'la...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#66 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571f8750, locals=0x0, con=0x7ffff43ca210, tstate=0x555555b67000) at ../Python/ceval.c:5065
#67 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571f8750, func=<function at remote 0x7ffff43ca200>) at ../Objects/call.c:342
#68 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571f8750, callable=<function at remote 0x7ffff43ca200>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#69 method_vectorcall (method=<optimized out>, args=0x5555571f8758, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#70 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f8758, callable=<method at remote 0x7fff885d4f80>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#71 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f8758, callable=<method at remote 0x7fff885d4f80>) at ../Include/cpython/abstract.h:123
#72 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff8470, tstate=<optimized out>) at ../Python/ceval.c:5891
#73 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#74 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571f85e0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 483, in _close_application_signal (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-python'...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#75 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571fedd8, locals=0x0, con=0x7fffe9f9acc0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#76 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571fedd8, func=<function at remote 0x7fffe9f9acb0>) at ../Objects/call.c:342
#77 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571fedd8, callable=<function at remote 0x7fffe9f9acb0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#78 method_vectorcall (method=<optimized out>, args=0x5555571fede0, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#79 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571fede0, callable=<method at remote 0x7fff885602c0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#80 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571fede0, callable=<method at remote 0x7fff885602c0>) at ../Include/cpython/abstract.h:123
#81 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff86a0, tstate=<optimized out>) at ../Python/ceval.c:5891
#82 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#83 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571fec60, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 480, in _close_application_shutdown_callback (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applicati...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#84 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x555557201fb0, locals=0x0, con=0x7fffe9f9ac30, tstate=0x555555b67000) at ../Python/ceval.c:5065
#85 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x555557201fb0, func=<function at remote 0x7fffe9f9ac20>) at ../Objects/call.c:342
#86 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x555557201fb0, callable=<function at remote 0x7fffe9f9ac20>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#87 method_vectorcall (method=<optimized out>, args=0x555557201fb8, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#88 0x0000555555699785 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x555557201fb8, callable=<method at remote 0x7fff885d51c0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#89 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x555557201fb8, callable=<method at remote 0x7fff885d51c0>) at ../Include/cpython/abstract.h:123
#90 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff88d0, tstate=<optimized out>) at ../Python/ceval.c:5891
#91 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4213
#92 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x555557201e40, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_handler.py, line 150, in emit_shutdown_plugin_completed (self=<PluginHandlerDirect(_main_window=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, _instance_id=<PluginInstanceId(plugin_id='rqt_graph/RosGraph', serial_number=0) at remote 0x7fffe03b85b0>, _application_context=<ApplicationContext(provide_app_dbus_interfaces=True, dbus_base_bus_name='org.ros.qt_gu...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#93 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571e5a90, locals=0x0, con=0x7fffe9f76840, tstate=0x555555b67000) at ../Python/ceval.c:5065
#94 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571e5a90, func=<function at remote 0x7fffe9f76830>) at ../Objects/call.c:342
#95 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571e5a90, callable=<function at remote 0x7fffe9f76830>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#96 method_vectorcall (method=<optimized out>, args=0x5555571e5a98, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#97 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571e5a98, callable=<method at remote 0x7fff885d5140>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#98 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571e5a98, callable=<method at remote 0x7fff885d5140>) at ../Include/cpython/abstract.h:123
#99 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff8b00, tstate=<optimized out>) at ../Python/ceval.c:5891
#100 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#101 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571e5920, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_handler_direct.py, line 92, in _shutdown_plugin (self=<PluginHandlerDirect(_main_window=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, _instance_id=<PluginInstanceId(plugin_id='rqt_graph/RosGraph', serial_number=0) at remote 0x7fffe03b85b0>, _application_context=<ApplicationContext(provide_app_dbus_interfaces=True, dbus_base_bus_name='org.ros.qt_gui', dbus...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#102 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571f2a18, locals=0x0, con=0x7fffe9f98950, tstate=0x555555b67000) at ../Python/ceval.c:5065
#103 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571f2a18, func=<function at remote 0x7fffe9f98940>) at ../Objects/call.c:342
#104 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571f2a18, callable=<function at remote 0x7fffe9f98940>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#105 method_vectorcall (method=<optimized out>, args=0x5555571f2a20, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#106 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f2a20, callable=<method at remote 0x7fff88560500>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#107 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f2a20, callable=<method at remote 0x7fff88560500>) at ../Include/cpython/abstract.h:123
#108 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff8d30, tstate=<optimized out>) at ../Python/ceval.c:5891
#109 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#110 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571f28a0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_handler.py, line 136, in shutdown_plugin (self=<PluginHandlerDirect(_main_window=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, _instance_id=<PluginInstanceId(plugin_id='rqt_graph/RosGraph', serial_number=0) at remote 0x7fffe03b85b0>, _application_context=<ApplicationContext(provide_app_dbus_interfaces=True, dbus_base_bus_name='org.ros.qt_gui', dbus_unique...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#111 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571f6900, locals=0x0, con=0x7fffe9f76720, tstate=0x555555b67000) at ../Python/ceval.c:5065
#112 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571f6900, func=<function at remote 0x7fffe9f76710>) at ../Objects/call.c:342
#113 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571f6900, callable=<function at remote 0x7fffe9f76710>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#114 method_vectorcall (method=<optimized out>, args=0x5555571f6908, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#115 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f6908, callable=<method at remote 0x7fff885d4fc0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#116 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f6908, callable=<method at remote 0x7fff885d4fc0>) at ../Include/cpython/abstract.h:123
#117 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff8f60, tstate=<optimized out>) at ../Python/ceval.c:5891
#118 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#119 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571f6780, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_handler_direct.py, line 80, in shutdown_plugin (self=<PluginHandlerDirect(_main_window=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, _instance_id=<PluginInstanceId(plugin_id='rqt_graph/RosGraph', serial_number=0) at remote 0x7fffe03b85b0>, _application_context=<ApplicationContext(provide_app_dbus_interfaces=True, dbus_base_bus_name='org.ros.qt_gui', dbus_...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#120 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571ff770, locals=0x0, con=0x7fffe9f988c0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#121 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571ff770, func=<function at remote 0x7fffe9f988b0>) at ../Objects/call.c:342
#122 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571ff770, callable=<function at remote 0x7fffe9f988b0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#123 method_vectorcall (method=<optimized out>, args=0x5555571ff778, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#124 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571ff778, callable=<method at remote 0x7fff88561900>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#125 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571ff778, callable=<method at remote 0x7fff88561900>) at ../Include/cpython/abstract.h:123
#126 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff9190, tstate=<optimized out>) at ../Python/ceval.c:5891
#127 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#128 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571ff5e0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 356, in _shutdown_plugin (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-python', 'iconty...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#129 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571ffc20, locals=0x0, con=0x7fffe9f9a180, tstate=0x555555b67000) at ../Python/ceval.c:5065
#130 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571ffc20, func=<function at remote 0x7fffe9f9a170>) at ../Objects/call.c:342
#131 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571ffc20, callable=<function at remote 0x7fffe9f9a170>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#132 method_vectorcall (method=<optimized out>, args=0x5555571ffc28, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#133 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571ffc28, callable=<method at remote 0x7fff885d5000>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#134 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571ffc28, callable=<method at remote 0x7fff885d5000>) at ../Include/cpython/abstract.h:123
#135 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff93c0, tstate=<optimized out>) at ../Python/ceval.c:5891
#136 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#137 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571ffaa0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 469, in _close_application_shutdown_plugins (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applicatio...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#138 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x555557201638, locals=0x0, con=0x7fffe9f9aba0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#139 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x555557201638, func=<function at remote 0x7fffe9f9ab90>) at ../Objects/call.c:342
#140 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x555557201638, callable=<function at remote 0x7fffe9f9ab90>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#141 method_vectorcall (method=<optimized out>, args=0x555557201640, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#142 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x555557201640, callable=<method at remote 0x7fff88563000>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#143 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x555557201640, callable=<method at remote 0x7fff88563000>) at ../Include/cpython/abstract.h:123
#144 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff95f0, tstate=<optimized out>) at ../Python/ceval.c:5891
#145 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#146 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555572014c0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 462, in _close_application_save_callback (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#147 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555571e9f70, locals=0x0, con=0x7fffe9f9ab10, tstate=0x555555b67000) at ../Python/ceval.c:5065
#148 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555571e9f70, func=<function at remote 0x7fffe9f9ab00>) at ../Objects/call.c:342
#149 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555571e9f70, callable=<function at remote 0x7fffe9f9ab00>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#150 method_vectorcall (method=<optimized out>, args=0x5555571e9f78, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#151 0x0000555555699785 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571e9f78, callable=<method at remote 0x7fff885d5240>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#152 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571e9f78, callable=<method at remote 0x7fff885d5240>) at ../Include/cpython/abstract.h:123
#153 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff9820, tstate=<optimized out>) at ../Python/ceval.c:5891
#154 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4213
#155 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571e9e00, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_handler.py, line 207, in emit_save_settings_completed (self=<PluginHandlerDirect(_main_window=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, _instance_id=<PluginInstanceId(plugin_id='rqt_graph/RosGraph', serial_number=0) at remote 0x7fffe03b85b0>, _application_context=<ApplicationContext(provide_app_dbus_interfaces=True, dbus_base_bus_name='org.ros.qt_gui'...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#156 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555572005b0, locals=0x0, con=0x7fffe9f76cc0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#157 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555572005b0, func=<function at remote 0x7fffe9f76cb0>) at ../Objects/call.c:342
#158 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555572005b0, callable=<function at remote 0x7fffe9f76cb0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#159 method_vectorcall (method=<optimized out>, args=0x5555572005b8, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#160 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555572005b8, callable=<method at remote 0x7fff885d5340>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#161 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555572005b8, callable=<method at remote 0x7fff885d5340>) at ../Include/cpython/abstract.h:123
#162 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff9a50, tstate=<optimized out>) at ../Python/ceval.c:5891
#163 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#164 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x555557200420, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_handler_direct.py, line 114, in _save_settings (self=<PluginHandlerDirect(_main_window=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, _instance_id=<PluginInstanceId(plugin_id='rqt_graph/RosGraph', serial_number=0) at remote 0x7fffe03b85b0>, _application_context=<ApplicationContext(provide_app_dbus_interfaces=True, dbus_base_bus_name='org.ros.qt_gui', dbus_...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#165 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x55555721de88, locals=0x0, con=0x7fffe9f98b00, tstate=0x555555b67000) at ../Python/ceval.c:5065
#166 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x55555721de88, func=<function at remote 0x7fffe9f98af0>) at ../Objects/call.c:342
#167 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x55555721de88, callable=<function at remote 0x7fffe9f98af0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#168 method_vectorcall (method=<optimized out>, args=0x55555721de90, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#169 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x55555721de90, callable=<method at remote 0x7fff885d5300>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#170 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x55555721de90, callable=<method at remote 0x7fff885d5300>) at ../Include/cpython/abstract.h:123
#171 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff9c80, tstate=<optimized out>) at ../Python/ceval.c:5891
#172 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#173 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x55555721dd00, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_handler.py, line 191, in save_settings (self=<PluginHandlerDirect(_main_window=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, _instance_id=<PluginInstanceId(plugin_id='rqt_graph/RosGraph', serial_number=0) at remote 0x7fffe03b85b0>, _application_context=<ApplicationContext(provide_app_dbus_interfaces=True, dbus_base_bus_name='org.ros.qt_gui', dbus_unique_b...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#174 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x555557200360, locals=0x0, con=0x7fffe9f76ba0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#175 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x555557200360, func=<function at remote 0x7fffe9f76b90>) at ../Objects/call.c:342
#176 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x555557200360, callable=<function at remote 0x7fffe9f76b90>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#177 method_vectorcall (method=<optimized out>, args=0x555557200368, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#178 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x555557200368, callable=<method at remote 0x7fff885d5100>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#179 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x555557200368, callable=<method at remote 0x7fff885d5100>) at ../Include/cpython/abstract.h:123
#180 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffff9eb0, tstate=<optimized out>) at ../Python/ceval.c:5891
#181 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#182 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555572001c0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 343, in _save_plugin_settings (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-python', 'i...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#183 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x555557202238, locals=0x0, con=0x7fffe9f9a060, tstate=0x555555b67000) at ../Python/ceval.c:5065
#184 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x555557202238, func=<function at remote 0x7fffe9f9a050>) at ../Objects/call.c:342
#185 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x555557202238, callable=<function at remote 0x7fffe9f9a050>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#186 method_vectorcall (method=<optimized out>, args=0x555557202240, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#187 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x555557202240, callable=<method at remote 0x7fff885d5280>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#188 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x555557202240, callable=<method at remote 0x7fff885d5280>) at ../Include/cpython/abstract.h:123
#189 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffa0e0, tstate=<optimized out>) at ../Python/ceval.c:5891
#190 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#191 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555572020a0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 432, in _save_settings (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-python', 'icontype...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#192 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x5555572013e0, locals=0x0, con=0x7fffe9f9a8d0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#193 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x5555572013e0, func=<function at remote 0x7fffe9f9a8c0>) at ../Objects/call.c:342
#194 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x5555572013e0, callable=<function at remote 0x7fffe9f9a8c0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#195 method_vectorcall (method=<optimized out>, args=0x5555572013e8, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#196 0x000055555569f462 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555572013e8, callable=<method at remote 0x7fff885d5500>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#197 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555572013e8, callable=<method at remote 0x7fff885d5500>) at ../Include/cpython/abstract.h:123
#198 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffa310, tstate=<optimized out>) at ../Python/ceval.c:5891
#199 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#200 0x00005555556bed7e in _PyEval_EvalFrame (throwflag=0, f=Frame 0x555557201260, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/plugin_manager.py, line 456, in close_application (self=<PluginManager(_plugin_provider=<CompositePluginProvider(_plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-python', 'icont...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#201 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x7fffffffa3c0, locals=0x0, con=0x7fffe9f9aa80, tstate=0x555555b67000) at ../Python/ceval.c:5065
#202 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x7fffffffa3c0, func=<function at remote 0x7fffe9f9aa70>) at ../Objects/call.c:342
#203 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x7fffffffa3c0, callable=<function at remote 0x7fffe9f9aa70>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#204 method_vectorcall (method=<optimized out>, args=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:83
#205 0x00007ffff5217ba0 in  () at /usr/lib/python3/dist-packages/PyQt5/QtCore.abi3.so
#206 0x00007ffff521b5ae in  () at /usr/lib/python3/dist-packages/PyQt5/QtCore.abi3.so
#207 0x00007ffff4cf14e5 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#208 0x00007ffff5216a4e in  () at /usr/lib/python3/dist-packages/PyQt5/QtCore.abi3.so
#209 0x00005555556951f0 in method_vectorcall_VARARGS (func=<optimized out>, args=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/descrobject.c:311
#210 0x00005555556998cb in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f2ed8, callable=<method_descriptor at remote 0x7ffff436a700>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#211 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555571f2ed8, callable=<method_descriptor at remote 0x7ffff436a700>) at ../Include/cpython/abstract.h:123
#212 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffa820, tstate=<optimized out>) at ../Python/ceval.c:5891
#213 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4198
#214 0x00005555556bed7e in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571f2d60, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/main_window.py, line 87, in closeEvent (self=<MainWindow(_help_widget=<QTextBrowser at remote 0x7fffe9f9b400>, _save_on_close_signaled=True, _global_settings=<Settings(_settings_proxy=<SettingsProxy(_qsettings=<QSettings at remote 0x7fffe9f9b2e0>, _mutex=<QMutex at remote 0x7fffe9f82570>) at remote 0x7fffea011d80>, _group='global') at remote 0x7fffea011ea0>, _perspective_settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph') at remote 0x7fffea013d90>, _settings=<Settings(_settings_proxy=<...>, _group='perspective/@rqt_graph__RosGraph/mainwindow') at remote 0x7fffe03b4af0>) at remote 0x7fffe9f9b370>, event=<QCloseEvent at remote 0x7fff8a7db400>), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#215 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x7fffffffa8d0, locals=0x0, con=0x7fffe9f61400, tstate=0x555555b67000) at ../Python/ceval.c:5065
#216 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x7fffffffa8d0, func=<function at remote 0x7fffe9f613f0>) at ../Objects/call.c:342
#217 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x7fffffffa8d0, callable=<function at remote 0x7fffe9f613f0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#218 method_vectorcall (method=<optimized out>, args=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:83
#219 0x00007ffff58745aa in  () at /usr/lib/python3/dist-packages/PyQt5/sip.cpython-310-x86_64-linux-gnu.so
#220 0x00007ffff5874693 in  () at /usr/lib/python3/dist-packages/PyQt5/sip.cpython-310-x86_64-linux-gnu.so
#221 0x00007ffff145be2a in  () at /usr/lib/python3/dist-packages/PyQt5/QtWidgets.abi3.so
#222 0x00007ffff0baf4ee in QWidget::event(QEvent*) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#223 0x00007ffff145d9eb in  () at /usr/lib/python3/dist-packages/PyQt5/QtWidgets.abi3.so
#224 0x00007ffff0b6c713 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#225 0x00007ffff15a0f16 in  () at /usr/lib/python3/dist-packages/PyQt5/QtWidgets.abi3.so
#226 0x00007ffff4cb9e3a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#227 0x00007ffff0baa58e in QWidgetPrivate::close_helper(QWidgetPrivate::CloseMode) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#228 0x00007ffff15d225f in  () at /usr/lib/python3/dist-packages/PyQt5/QtWidgets.abi3.so
#229 0x00005555556b09c8 in cfunction_call (func=<built-in method close of MainWindow object at remote 0x7fffe9f9b370>, args=<optimized out>, kwargs=<optimized out>) at ../Objects/methodobject.c:552
#230 0x00005555556a74ab in _PyObject_MakeTpCall (tstate=0x555555b67000, callable=<built-in method close of MainWindow object at remote 0x7fffe9f9b370>, args=<optimized out>, nargs=<optimized out>, keywords=0x0) at ../Objects/call.c:215
#231 0x000055555569fe66 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=<optimized out>, callable=<built-in method close of MainWindow object at remote 0x7fffe9f9b370>, tstate=<optimized out>) at ../Include/cpython/abstract.h:112
#232 _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x555557201b08, callable=<built-in method close of MainWindow object at remote 0x7fffe9f9b370>, tstate=<optimized out>) at ../Include/cpython/abstract.h:99
#233 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x555557201b08, callable=<built-in method close of MainWindow object at remote 0x7fffe9f9b370>) at ../Include/cpython/abstract.h:123
#234 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffaeb0, tstate=<optimized out>) at ../Python/ceval.c:5891
#235 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#236 0x00005555556b11ec in _PyEval_EvalFrame (throwflag=0, f=Frame 0x555557201980, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/main.py, line 463, in sigint_handler (args=(2, Frame 0x5555571f00c0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/main.py, line 468, in <lambda> ())), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#237 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, con=0x7fffe9f9b530, tstate=0x555555b67000) at ../Python/ceval.c:5065
#238 _PyFunction_Vectorcall (func=<function at remote 0x7fffe9f9b520>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/call.c:342
#239 0x0000555555790a3a in _PyErr_CheckSignalsTstate (tstate=0x555555b67000) at ../Modules/signalmodule.c:1847
#240 0x00005555557b75d3 in handle_signals (tstate=0x555555b67000) at ../Python/ceval.c:692
#241 0x00005555556a1bb2 in eval_frame_handle_pending (tstate=<optimized out>) at ../Python/ceval.c:1166
#242 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:1775
#243 0x00005555556b11ec in _PyEval_EvalFrame (throwflag=0, f=Frame 0x5555571f00c0, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/main.py, line 468, in <lambda> (), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#244 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, con=0x7fffe9f9b650, tstate=0x555555b67000) at ../Python/ceval.c:5065
#245 _PyFunction_Vectorcall (func=<function at remote 0x7fffe9f9b640>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/call.c:342
#246 0x00007ffff5217ba0 in  () at /usr/lib/python3/dist-packages/PyQt5/QtCore.abi3.so
#247 0x00007ffff521b5ae in  () at /usr/lib/python3/dist-packages/PyQt5/QtCore.abi3.so
#248 0x00007ffff4cf14e5 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#249 0x00007ffff4cf57fe in QTimer::timeout(QTimer::QPrivateSignal) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#250 0x00007ffff51059b3 in  () at /usr/lib/python3/dist-packages/PyQt5/QtCore.abi3.so
#251 0x00007ffff4ce733f in QObject::event(QEvent*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#252 0x00007ffff50ff933 in  () at /usr/lib/python3/dist-packages/PyQt5/QtCore.abi3.so
#253 0x00007ffff0b6c713 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#254 0x00007ffff15a0f16 in  () at /usr/lib/python3/dist-packages/PyQt5/QtWidgets.abi3.so
#255 0x00007ffff4cb9e3a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#256 0x00007ffff4d123eb in QTimerInfoList::activateTimers() () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#257 0x00007ffff4d12d34 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#258 0x00007ffff541dd3b in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#259 0x00007ffff54726c8 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#260 0x00007ffff541b3e3 in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#261 0x00007ffff4d130b8 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#262 0x00007ffff4cb875b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#263 0x00007ffff4cc0cf4 in QCoreApplication::exec() () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#264 0x00007ffff159b9ee in  () at /usr/lib/python3/dist-packages/PyQt5/QtWidgets.abi3.so
#265 0x00005555556b09c8 in cfunction_call (func=<built-in method exec_ of QApplication object at remote 0x7fffe9f74c10>, args=<optimized out>, kwargs=<optimized out>) at ../Objects/methodobject.c:552
#266 0x00005555556a74ab in _PyObject_MakeTpCall (tstate=0x555555b67000, callable=<built-in method exec_ of QApplication object at remote 0x7fffe9f74c10>, args=<optimized out>, nargs=<optimized out>, keywords=0x0) at ../Objects/call.c:215
#267 0x000055555569fe66 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=<optimized out>, callable=<built-in method exec_ of QApplication object at remote 0x7fffe9f74c10>, tstate=<optimized out>) at ../Include/cpython/abstract.h:112
#268 _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x555555e26b60, callable=<built-in method exec_ of QApplication object at remote 0x7fffe9f74c10>, tstate=<optimized out>) at ../Include/cpython/abstract.h:99
#269 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x555555e26b60, callable=<built-in method exec_ of QApplication object at remote 0x7fffe9f74c10>) at ../Include/cpython/abstract.h:123
#270 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffb9e0, tstate=<optimized out>) at ../Python/ceval.c:5891
#271 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4181
#272 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x555555e26750, for file /home/chenlh/Projects/ROS2/ros2-master/install/lib/python3.10/site-packages/qt_gui/main.py, line 650, in main (argv=['/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph'], standalone='rqt_graph.ros_graph.RosGraph', plugin_argument_provider=None, plugin_manager_settings_prefix='', help_text='<p><b>rqt</b> is a GUI framework that is able to load various plug-in tools\nas dockable windows. There are currently no plug-ins selected. To add plug-ins, select items from\nthe <b>Plugins</b> menu.</p>\n<p>You may also save a particular arrangement of plug-ins as a <i>perspective</i> using the\n<b>Perspectives</b> menu.\n', arguments=[], plugin_args=[], parser=<ArgumentParser(description=None, argument_default=None, prefix_chars='-', conflict_handler='error', _registries={'action': {None: <type at remote 0x555555d16be0>, 'store': <type at remote 0x555555d16be0>, 'store_const': <type at remote 0x555555d16fa0>, 'store_true': <type at remote 0x555555d17360>, 'store_false': <type at ...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#273 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x7ffff6c435d0, locals=0x0, con=0x7ffff69efad0, tstate=0x555555b67000) at ../Python/ceval.c:5065
#274 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x7ffff6c435d0, func=<function at remote 0x7ffff69efac0>) at ../Objects/call.c:342
#275 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x7ffff6c435d0, callable=<function at remote 0x7ffff69efac0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#276 method_vectorcall (method=<optimized out>, args=0x7ffff6c435d8, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#277 0x000055555569aadf in _PyObject_VectorcallTstate (kwnames=('standalone', 'plugin_argument_provider', 'help_text'), nargsf=<optimized out>, args=<optimized out>, callable=<method at remote 0x7ffff588cac0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#278 PyObject_Vectorcall (kwnames=('standalone', 'plugin_argument_provider', 'help_text'), nargsf=<optimized out>, args=<optimized out>, callable=<method at remote 0x7ffff588cac0>) at ../Include/cpython/abstract.h:123
#279 call_function (kwnames=('standalone', 'plugin_argument_provider', 'help_text'), oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffbc10, tstate=<optimized out>) at ../Python/ceval.c:5891
#280 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4231
#281 0x00005555556bec31 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x7ffff6c43440, for file /home/chenlh/Projects/ROS2/ros2-master/build/rqt_gui/src/rqt_gui/main.py, line 62, in main (self=<Main(_qtgui_path='/home/chenlh/Projects/ROS2/ros2-master/install', _settings_filename='rqt_gui', plugin_providers=[<RospkgPluginProvider(_export_tag='qt_gui', _base_class_type='qt_gui_py::Plugin', _plugin_descriptors={'rqt_py_console/PyConsole': <PluginDescriptor(_dict={'plugin_id': 'rqt_py_console/PyConsole', 'attributes': {'package_name': 'rqt_py_console', 'plugin_path': '/home/chenlh/Projects/ROS2/ros2-master/install/share/rqt_py_console', 'library_path': 'src', 'class_name': 'PyConsole', 'class_type': 'rqt_py_console.py_console.PyConsole', 'class_base_class_type': 'qt_gui_py::Plugin', 'plugin_id': 'rqt_py_console/PyConsole', 'module_name': 'rqt_py_console.py_console', 'class_from_class_type': 'PyConsole', 'not_available': ''}, 'action': {'label': 'Python Console', 'statustip': 'A Python GUI plugin providing an interactive Python console.', 'icon': 'applications-python', 'icontyp...(truncated), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#282 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=0x7ffff6d9b810, locals=0x0, con=0x7ffff697af00, tstate=0x555555b67000) at ../Python/ceval.c:5065
#283 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x7ffff6d9b810, func=<function at remote 0x7ffff697aef0>) at ../Objects/call.c:342
#284 _PyObject_VectorcallTstate (kwnames=<optimized out>, nargsf=<optimized out>, args=0x7ffff6d9b810, callable=<function at remote 0x7ffff697aef0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#285 method_vectorcall (method=<optimized out>, args=0x7ffff6d9b818, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/classobject.c:53
#286 0x000055555569aadf in _PyObject_VectorcallTstate (kwnames=('standalone',), nargsf=<optimized out>, args=<optimized out>, callable=<method at remote 0x7ffff58949c0>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#287 PyObject_Vectorcall (kwnames=('standalone',), nargsf=<optimized out>, args=<optimized out>, callable=<method at remote 0x7ffff58949c0>) at ../Include/cpython/abstract.h:123
#288 call_function (kwnames=('standalone',), oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffbe40, tstate=<optimized out>) at ../Python/ceval.c:5891
#289 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4231
#290 0x00005555556b11ec in _PyEval_EvalFrame (throwflag=0, f=Frame 0x7ffff6d9b6a0, for file /home/chenlh/Projects/ROS2/ros2-master/build/rqt_graph/src/rqt_graph/main.py, line 9, in main (), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#291 _PyEval_Vector (kwnames=<optimized out>, argcount=<optimized out>, args=<optimized out>, locals=0x0, con=0x7ffff6b87410, tstate=0x555555b67000) at ../Python/ceval.c:5065
#292 _PyFunction_Vectorcall (func=<function at remote 0x7ffff6b87400>, stack=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>) at ../Objects/call.c:342
#293 0x0000555555699785 in _PyObject_VectorcallTstate (kwnames=0x0, nargsf=<optimized out>, args=0x7ffff6e3f9b8, callable=<function at remote 0x7ffff6b87400>, tstate=0x555555b67000) at ../Include/cpython/abstract.h:114
#294 PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x7ffff6e3f9b8, callable=<function at remote 0x7ffff6b87400>) at ../Include/cpython/abstract.h:123
#295 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, trace_info=0x7fffffffc010, tstate=<optimized out>) at ../Python/ceval.c:5891
#296 _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:4213
#297 0x0000555555695ed6 in _PyEval_EvalFrame (throwflag=0, f=Frame 0x7ffff6e3f840, for file /home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph, line 33, in <module> (), tstate=0x555555b67000) at ../Include/internal/pycore_ceval.h:46
#298 _PyEval_Vector (tstate=0x555555b67000, con=<optimized out>, locals=<optimized out>, args=<optimized out>, argcount=<optimized out>, kwnames=<optimized out>) at ../Python/ceval.c:5065
#299 0x000055555578c366 in PyEval_EvalCode (co=<code at remote 0x7ffff6d31000>, globals={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph') at remote 0x7ffff6e1fe80>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7ffff6f905e0>, '__file__': '/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', '__cached__': None, 're': <module at remote 0x7ffff6d4e2f0>, 'sys': <module at remote 0x7ffff6f7e250>, '__requires__': 'rqt-graph', 'distribution': <function at remote 0x7ffff6d8ec20>, 'importlib_load_entry_point': <function at remote 0x7ffff6f63f40>, 'load_entry_point': <function at remote 0x7ffff6f63f40>}, locals=<optimized out>) at ../Python/ceval.c:1134
#300 0x00005555557b9108 in run_eval_code_obj (tstate=0x555555b67000, co=0x7ffff6d31000, globals={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph') at remote 0x7ffff6e1fe80>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7ffff6f905e0>, '__file__': '/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', '__cached__': None, 're': <module at remote 0x7ffff6d4e2f0>, 'sys': <module at remote 0x7ffff6f7e250>, '__requires__': 'rqt-graph', 'distribution': <function at remote 0x7ffff6d8ec20>, 'importlib_load_entry_point': <function at remote 0x7ffff6f63f40>, 'load_entry_point': <function at remote 0x7ffff6f63f40>}, locals={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph') at remote 0x7ffff6e1fe80>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7ffff6f905e0>, '__file__': '/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', '__cached__': None, 're': <module at remote 0x7ffff6d4e2f0>, 'sys': <module at remote 0x7ffff6f7e250>, '__requires__': 'rqt-graph', 'distribution': <function at remote 0x7ffff6d8ec20>, 'importlib_load_entry_point': <function at remote 0x7ffff6f63f40>, 'load_entry_point': <function at remote 0x7ffff6f63f40>}) at ../Python/pythonrun.c:1291
#301 0x00005555557b1f5b in run_mod (mod=<optimized out>, filename=<optimized out>, globals={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph') at remote 0x7ffff6e1fe80>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7ffff6f905e0>, '__file__': '/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', '__cached__': None, 're': <module at remote 0x7ffff6d4e2f0>, 'sys': <module at remote 0x7ffff6f7e250>, '__requires__': 'rqt-graph', 'distribution': <function at remote 0x7ffff6d8ec20>, 'importlib_load_entry_point': <function at remote 0x7ffff6f63f40>, 'load_entry_point': <function at remote 0x7ffff6f63f40>}, locals={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph') at remote 0x7ffff6e1fe80>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7ffff6f905e0>, '__file__': '/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', '__cached__': None, 're': <module at remote 0x7ffff6d4e2f0>, 'sys': <module at remote 0x7ffff6f7e250>, '__requires__': 'rqt-graph', 'distribution': <function at remote 0x7ffff6d8ec20>, 'importlib_load_entry_point': <function at remote 0x7ffff6f63f40>, 'load_entry_point': <function at remote 0x7ffff6f63f40>}, flags=<optimized out>, arena=<optimized out>) at ../Python/pythonrun.c:1312
#302 0x00005555557b8e55 in pyrun_file (fp=fp@entry=0x555555b45050, filename=filename@entry='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', start=start@entry=257, globals=globals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph') at remote 0x7ffff6e1fe80>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7ffff6f905e0>, '__file__': '/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', '__cached__': None, 're': <module at remote 0x7ffff6d4e2f0>, 'sys': <module at remote 0x7ffff6f7e250>, '__requires__': 'rqt-graph', 'distribution': <function at remote 0x7ffff6d8ec20>, 'importlib_load_entry_point': <function at remote 0x7ffff6f63f40>, 'load_entry_point': <function at remote 0x7ffff6f63f40>}, locals=locals@entry={'__name__': '__main__', '__doc__': None, '__package__': None, '__loader__': <SourceFileLoader(name='__main__', path='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph') at remote 0x7ffff6e1fe80>, '__spec__': None, '__annotations__': {}, '__builtins__': <module at remote 0x7ffff6f905e0>, '__file__': '/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', '__cached__': None, 're': <module at remote 0x7ffff6d4e2f0>, 'sys': <module at remote 0x7ffff6f7e250>, '__requires__': 'rqt-graph', 'distribution': <function at remote 0x7ffff6d8ec20>, 'importlib_load_entry_point': <function at remote 0x7ffff6f63f40>, 'load_entry_point': <function at remote 0x7ffff6f63f40>}, closeit=closeit@entry=1, flags=0x7fffffffc2b8) at ../Python/pythonrun.c:1208
#303 0x00005555557b8338 in _PyRun_SimpleFileObject (fp=0x555555b45050, filename='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', closeit=1, flags=0x7fffffffc2b8) at ../Python/pythonrun.c:456
#304 0x00005555557b8033 in _PyRun_AnyFileObject (fp=0x555555b45050, filename='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', closeit=1, flags=0x7fffffffc2b8) at ../Python/pythonrun.c:90
#305 0x00005555557a92de in pymain_run_file_obj (skip_source_first_line=0, filename='/home/chenlh/Projects/ROS2/ros2-master/install/bin/rqt_graph', program_name='/usr/bin/python3') at ../Modules/main.c:353
#306 pymain_run_file (config=0x555555b45b40) at ../Modules/main.c:372
#307 pymain_run_python (exitcode=0x7fffffffc2b4) at ../Modules/main.c:587
#308 Py_RunMain () at ../Modules/main.c:666
#309 0x000055555577f32d in Py_BytesMain (argc=<optimized out>, argv=<optimized out>) at ../Modules/main.c:720
#310 0x00007ffff7c29d90 in __libc_start_call_main (main=main@entry=0x55555577f2f0 <main>, argc=argc@entry=2, argv=argv@entry=0x7fffffffc4c8) at ../sysdeps/nptl/libc_start_call_main.h:58
#311 0x00007ffff7c29e40 in __libc_start_main_impl (main=0x55555577f2f0 <main>, argc=2, argv=0x7fffffffc4c8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffc4b8) at ../csu/libc-start.c:392
#312 0x000055555577f225 in _start ()
(gdb) 


```

</p>
</details>

`rclpy.shutdown()` is called in thread 26, so the [QThread](https://github.com/ros-visualization/rqt/blob/3f0f2054bbff23e88b1d24a00171014e238d8c45/rqt_gui_py/src/rqt_gui_py/rclpy_spinner.py#L31) can exit normally.

In thread 1, we could see the `shutdown` for the `RosPyPluginProvider`
```
shutdown 
  ->  self._spinner.quit  ( Too late to ask the thread to exit. Sorry, I didn't add a breakpoint here.)
  ->  _destroy_node ( the backtrace above goes this path )
```


So I think `rclpy.shutdown() was called before QThread.quit()` is a normal case.

I'll close this PR if `general strategy` is implemented.

Reply:
@Yadunund This is more or less the fix that we discussed.

I think that the warning here is more noise than helpful.  As @iuhilnehc-ynos states, When using ctrl-c in the terminal or sending signals (via launch), there could be multiple cases where rclpy.shutdown is called before the qthread quit.

I would still like to address this in the larger, more general case, but that is likely going to break API and need to land in rolling with a potential iron backport.

For now, this will take away the unnecessarily-scary warnings when stopping rqt.

Reply:
@iuhilnehc-ynos thanks for the detailed explanation! Really appreciate it! That sounds good @mjcarroll